src/WebJobs.Extensions.DurableTask/DurabilityProvider.cs

```diff
@@ -374,5 +374,17 @@ public Task PurgeOrchestrationHistoryAsync(DateTime thresholdDateTimeUtc, Orches
         {
             return this.GetOrchestrationServiceClient().PurgeOrchestrationHistoryAsync(thresholdDateTimeUtc, timeRangeFilterType);
         }
+
+        /// <summary>
+        /// Uses durability provider specific logic to verify whether a timespan for a timer
+        /// or a retry interval is allowed by the provider.
+        /// </summary>
+        /// <param name="timespan">The timespan for a timer or retry interval.</param>
+        /// <returns>A boolean indicating whether the time interval is valid and an error message if it is invalid.</returns>
+        public virtual (bool, string) CheckTimeInterval(TimeSpan timespan)
```

Comment:
I would prefer we not use Tuples in our public APIs. If we need to return two values, let's either create a new type or use a return value and an `out` parameter. There is some interesting discussion of this topic [here](https://stackoverflow.com/questions/3017352/is-using-net-4-0-tuples-in-my-c-sharp-code-a-poor-design-decision).